lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -822,15 +876,64 @@ SessionConfig SessionState::get_config() const {
   return config_;
 }
 
-uint32_t SessionState::get_local_teid() const {
-  return local_teid_;
-}
-
-void SessionState::set_local_teid(
-    uint32_t teid, SessionStateUpdateCriteria& uc) {
-  local_teid_              = teid;
-  uc.is_local_teid_updated = true;
-  uc.local_teid_updated    = teid;
+uint32_t SessionState::get_upf_local_teid() const {
+  char teidar[10];
+  union mix_t {
+    std::uint32_t theDWord;
+    std::uint8_t theBytes[4];
+  };
+  mix_t aVal;
+  // Retreive value
+  std::string teids = config_.rat_specific_context.m5gsm_session_context()
+                          .upf_endpoint()
+                          .teid();
+  uint32_t len = teids.length();
+  memset(teidar, '\0', 10);
+  memcpy(&teidar[0], &teids[0], len);
```

Comment:
why are you using memset and memcopy here? Isn't this just a conversion 
```
char* chr = str;
```

Also why is the TEID stored in such a weird format. Shouldn't we take advantage of protobuf to send/store the value in a readable format and  only convert to 3gpp once it leaves magma?

Reply:
Your are right, for various reasons  few params not terminated at NAS level, those values carried as it is till SMF.  Now  we need to change this end to end, also AMF team have few concerns to change this to uint32_t

Reply:
Its requirement/need from AMF team they receive and sent in byte format. It  is interlinked to many things.